types.go

```diff
@@ -111,14 +119,19 @@ type Problem struct {
 	// Func evaluates the objective function at the given location. Func
 	// must not modify x.
 	Func func(x []float64) float64
+
```

Comment:
Good idea. There is always someone who wants to modify x with a good, but misguided intention.